devDocs/technicalDesignOverview.md

```diff
@@ -0,0 +1,253 @@
+These objects are derived from the `textInfos.TextInfo` base class.
+
+TextInfo objects contain properties and methods to:
+* Move or expand the range by units such as character, word, line and paragraph
+* compare the start and end of a range with itself or another range
+* Fetch the text and formatting of the range
+ 
+ You can fetch a TextInfo object from an NVDA object via its `makeTextInfo` method, passing in the particular `textInfos.POSITION_*` constant depending on whether you want to fetch a range representing the position of the caret, selection, start or end of the text, or the entire text.
+ 
+### Global Commands
+The global commands object (`globalCommands.GlobalCommands`) contains built-in global scripts; i.e.
+they can be executed everywhere.
+For example, the review, report current focus and date/time scripts are all located in global commands.
+
+### Plugins
+NVDA allows third-parties to extend NVDA's functionality through plugins and add-ons.
+These may define custom NVDA objects for specific applications, add global features and add support for new braille displays and speech synthesizers.
+There are three plugin types: appModules, globalPlugins and drivers, with drivers further divided between speech synthesizer and braille display support.
+
+#### App Modules
+Generally, most widgets may appear in any application and an [NVDA object](#nvda-objects) should therefore be included in the main `NVDAObjects` package.
+However, there are sometimes cases where a widget is implemented specifically for one application, as well as cases where a single event must be overridden or a script must be provided only in one application.
+An app module provides support specific to an application for these cases.
+An app module is derived from the `appModuleHandler.AppModule` base class.
+App modules receive events for all [NVDA objects](#nvda-objects) in the application and can bind scripts which can be executed anywhere in that application.
+They can also implement their own NVDA objects for use within the application.
+
+#### Global Plugins
+Aside from application specific customisation using [app modules](#app-modules), it is also possible to extend NVDA on a global level.
+For example, new global commands can be added, behaviour can be changed and new GUI toolkits can be supported.
+This can be done using global plugins.
+A global plugin is derived from the `globalPluginHandler.GlobalPlugin` base class.
+Similar to [global commands](#global-commands), they can bind scripts which can be executed everywhere.
+More specifically, global plugins receive events for all [NVDA objects](#nvda-objects) in the Operating System and can bind scripts which can be executed anywhere.
+They can also implement their own global [NVDA Objects](#nvda-objects).
+
+### Tree Interceptors
+Sometimes, it is necessary to intercept events and scripts for an entire hierarchy (or tree) of [NVDA objects](#nvda-objects).
```

Comment:
This leaves me wondering, intercept between which two points? Where is the event "coming from", and where is it "going"?